src/OpenTelemetry.Instrumentation.Process/ProcessMetrics.cs

```diff
@@ -72,32 +80,86 @@ public ProcessMetrics(ProcessInstrumentationOptions options)
                 {
-                    new Measurement<double>(userProcessorTimeValue, new KeyValuePair<string, object?>("state", "user")),
-                    new Measurement<double>(privilegedProcessorTimeValue, new KeyValuePair<string, object?>("state", "system")),
+                    new Measurement<double>(userProcessorTimeSecondsValue, new KeyValuePair<string, object?>("state", "user")),
+                    new Measurement<double>(privilegedProcessorTimeSecondsValue, new KeyValuePair<string, object?>("state", "system")),
                 };
             },
             unit: "s",
             description: "Total CPU seconds broken down by different states.");
+
+        this.MeterInstance.CreateObservableGauge(
+            "process.cpu.utilization",
+            () =>
+            {
+                if (this.cpuUtilization == null)
+                {
+                    this.Snapshot();
+                }
+
+                var value = this.cpuUtilization;
+                this.cpuUtilization = null;
+                return value;
+            },
+            unit: "1",
+            description: "Difference in process.cpu.time since the last measurement, divided by the elapsed time and number of CPUs available to the process.");
+
+        // TODO: change to ObservableUpDownCounter
+        this.MeterInstance.CreateObservableGauge(
+            "process.threads",
+            () =>
+            {
+                if (!this.numberOfThreads.HasValue)
+                {
+                    this.Snapshot();
+                }
+
+                var value = this.numberOfThreads.Value;
+                this.numberOfThreads = null;
+                return value;
+            },
+            unit: "{threads}",
+            description: "Process threads count.");
     }
 
     private void Snapshot()
     {
         this.currentProcess.Refresh();
-        this.memoryUsage = this.currentProcess.WorkingSet64;
-        this.virtualMemoryUsage = this.currentProcess.PrivateMemorySize64;
-        this.userProcessorTime = this.currentProcess.UserProcessorTime.TotalSeconds;
-        this.privilegedProcessorTime = this.currentProcess.PrivilegedProcessorTime.TotalSeconds;
+        this.memoryUsageBytes = this.currentProcess.WorkingSet64;
+        this.virtualMemoryUsageBytes = this.currentProcess.PrivateMemorySize64;
+        this.userProcessorTimeSeconds = this.currentProcess.UserProcessorTime.TotalSeconds;
+        this.privilegedProcessorTimeSeconds = this.currentProcess.PrivilegedProcessorTime.TotalSeconds;
+        this.cpuUtilization = this.GetCpuUtilization();
+        this.numberOfThreads = this.currentProcess.Threads.Count;
+    }
+
+    private IEnumerable<Measurement<double>> GetCpuUtilization()
+    {
+        var userProcessorUtilization = (this.userProcessorTimeSeconds - this.lastCollectedUserProcessorTime) /
+                ((DateTime.UtcNow - this.lastCollectedTimeStampUtc).TotalSeconds * Environment.ProcessorCount);
```

Comment:
It's probably a good idea to extract the denominator out into a separate variable. Although a rare case, the thread could get preempted, and have a slightly different value for `DateTime.UtcNow` for calculating `privilegedProcessorUtilization`.

Reply:
I'm having troubles coming up with the variable name for that.... :face_in_clouds:
"denominator", "computePowerWhenEngineAllon"?

Reply:
I ended up using "temp" rofl. Please let me know what do you think.

Reply:
updated to use "elapsedTimeForAllCpus" as suggested!